ArgumentValueResolver/PsrServerRequestResolver.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Symfony\Bridge\PsrHttpMessage\ArgumentValueResolver;
+
+use Psr\Http\Message\MessageInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Injects the RequestInterface, MessageInterface or ServerRequestInterface when requested.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+final class PsrServerRequestResolver implements ArgumentValueResolverInterface
+{
+    private const SUPPORTED_TYPES = [
+        ServerRequestInterface::class => true,
+        RequestInterface::class => true,
+        MessageInterface::class => true,
+    ];
+
+    private $httpMessageFactory;
+
+    public function __construct(HttpMessageFactoryInterface $httpMessageFactory)
```

Comment:
we could make the argument nullable, which would allow removing the `HttpFoundationFactoryInterface` service in the recipe

Reply:
but then we need to auto-detect which of the HttpMessageFactory implementation to use (which depends on the psr-7 implementation being used). I don't think that's something we want in the constructor

Reply:
Those two factories, `HttpMessageFactoryInterface` and `HttpFoundationFactoryInterface`, are usually the reason why people install this package in the first place. I don't see that we gain much by removing them from the recipe. In fact, if we think about making the service wiring more complex, I'm leaning towards the bundle idea again, I must admit.

Reply:
OK, these services are useful ones, for end-users. Good point.